src/CloudNative.CloudEvents.AspNetCore/HttpRequestExtensions.cs

```diff
@@ -87,8 +89,51 @@ public static class HttpRequestExtensions
             }
         }
 
+        /// <summary>
+        /// Converts this HTTP request into a batch of CloudEvents.
+        /// </summary>
+        /// <param name="httpRequest">The HTTP request to decode. Must not be null.</param>
+        /// <param name="formatter">The event formatter to use to process the request body. Must not be null.</param>
+        /// <param name="extensions">The extension attributes to use when populating the CloudEvent. May be null.</param>
+        /// <returns>The decoded CloudEvent.</returns>
+        /// <exception cref="ArgumentException">The request does not contain a CloudEvent.</exception>
+        public static ValueTask<IReadOnlyList<CloudEvent>> ToCloudEventBatchAsync(
+            this HttpRequest httpRequest,
+            CloudEventFormatter formatter,
+            params CloudEventAttribute[] extensionAttributes) =>
+            ToCloudEventBatchAsync(httpRequest, formatter, (IEnumerable<CloudEventAttribute>) extensionAttributes);
+
+        /// <summary>
+        /// Converts this HTTP request into a batch of CloudEvents.
+        /// </summary>
+        /// <param name="httpRequest">The HTTP request to decode. Must not be null.</param>
+        /// <param name="formatter">The event formatter to use to process the request body. Must not be null.</param>
+        /// <param name="extensions">The extension attributes to use when populating the CloudEvent. May be null.</param>
+        /// <returns>The decoded CloudEvent.</returns>
+        /// <exception cref="ArgumentException">The request does not contain a CloudEvent.</exception>
+        public static async ValueTask<IReadOnlyList<CloudEvent>> ToCloudEventBatchAsync(
```

Comment:
Why use ValueTask here? I know sometimes the method will not be completed asynchronously, but that is only if an exception is thrown. In other worse, you won't ever get back a list of CloudEvents synchronously (or at least you will be getting back a Task from formatter.DecodeBatchModeMessageAsync)

Reply:
Good question. Mostly because I was following the pattern for ToCloudEventAsync. I wonder whether we should make them both return Task instead. I'd like to be consistent between the two, at least.

I'll merge this PR as-is, but create an issue to remind us to discuss this.